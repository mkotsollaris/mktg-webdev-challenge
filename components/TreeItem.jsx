import { useState } from "react"

// maybe not needed
const TreeItem = ({ label, intend, children }) => {
    let cssIntend = intend * 30;
    return <div style={{ textIndent: `${cssIntend}px` }}>
        {label}
        {children}
    </div>
}

export default TreeItem;