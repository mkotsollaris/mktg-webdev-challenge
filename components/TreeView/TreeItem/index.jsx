import { useContext } from 'react'
import ArrowIcon from '../../../public/img/arrow.svg'
import style from './style.module.css'
import componentsStyle from '../../style.module.css'
import AppContext from '../../../context/AppContext'

const ExpandIconElement = (
  <img alt="expand" className={style.arrow} src={ArrowIcon} />
)
const CollapseIconElement = (
  <img
    alt="collapse"
    className={`${style['rotate-90']} ${style.arrow}`}
    src={ArrowIcon}
  />
)

const TreeItem = ({ label, children, level }) => {
  const {
    filteredDepartments,
    computeFilteredDepartments,
    expanded,
    setExpanded,
  } = useContext(AppContext)

  const isExpanded = () => {
    return expanded?.includes(label)
  }

  const onClick = () => {
    computeFilteredDepartments(label)

    if (isExpanded()) {
      setExpanded(expanded?.filter((e) => e !== label))
    } else {
      setExpanded([...expanded, label])
    }
  }

  const Icon =
    !children && level === 1 ? (
      <div />
    ) : !children && level > 1 ? null : isExpanded() ? (
      <div>{CollapseIconElement}</div>
    ) : (
      <div>{ExpandIconElement}</div>
    )

  const classNames = filteredDepartments?.includes(label)
    ? `${style.treeItem} ${style.selected}`
    : style.treeItem

  const element = (
    <div className={`${style.treeItem} ${componentsStyle['non-selectable']}`}>
      <button
        tabIndex={0}
        onClick={onClick}
        className={`${style.expandableTreeItem} ${classNames} `}
      >
        {Icon}
        {label}
      </button>
      {isExpanded() ? (
        <ul className={style.ul}>
          <div className={style.children}>{children}</div>
        </ul>
      ) : null}
    </div>
  )
  return element
}

export default TreeItem
