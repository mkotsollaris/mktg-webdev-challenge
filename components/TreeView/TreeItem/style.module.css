.expandableTreeItem {
  display: grid;
  align-content: center;
  justify-content: center;
  grid-template-columns: 10% 90%;
}

.treeItem {
  cursor: pointer;
  color: darkslategray;
  padding-top: 0.1rem;
  padding-bottom: 0.1rem;
  &.selected {
    color: dodgerblue;
  }
}

.ul {
  margin-left: 1.5rem;
  padding-inline-start: 1rem;
  margin-block-end: 0;
  margin-block-start: 0;
  border-left: 1px solid lightgray;
  padding: 0;
  & li {
    list-style: none;
  }
}

.arrow {
  margin-bottom: 0.1rem;
}

.rotate-90 {
  transform: rotate(90deg);
}
