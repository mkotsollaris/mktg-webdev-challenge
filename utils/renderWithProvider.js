import { render } from '@testing-library/react'
import AppContext from '../context/AppContext'

function renderWithProvider(props, Component) {
  return render(
    <AppContext.Provider value={props}>
      <Component />
    </AppContext.Provider>
  )
}

export default renderWithProvider
