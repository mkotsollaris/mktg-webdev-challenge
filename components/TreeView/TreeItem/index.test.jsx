import TreeItem from '.'
import renderWithProvider from '../../../utils/renderWithProvider'

it('renders', () => {
  const { asFragment } = renderWithProvider({}, TreeItem)
  expect(asFragment()).toMatchSnapshot()
})
