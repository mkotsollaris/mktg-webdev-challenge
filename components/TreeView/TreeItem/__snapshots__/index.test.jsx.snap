// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="undefined undefined"
  >
    <button
      class="undefined undefined "
      tabindex="0"
    >
      <div>
        <img
          alt="expand"
          src="[object Object]"
        />
      </div>
    </button>
  </div>
</DocumentFragment>
`;
